File: THCS/2023_2024/LaoCai_2324/Bai3/bai3.cpp
#include <bits/stdc++.h>
#define ll long long
#define el cout << '\n'
#define f0(i,n) for(ll i=0;i<n;i++)
using namespace std;

char* inp_file_name = "bai3";

int main(){
    ios_base::sync_with_stdio(0);cin.tie(0);cout.tie(0);

    if(inp_file_name != ""){
        char *s = new char[strlen(inp_file_name)+5];
        strcpy(s,inp_file_name);
        strcat(s, ".inp");
        freopen(s,"r",stdin);
        strcpy(s,inp_file_name);
        strcat(s, ".out");
        freopen(s,"w",stdout);
    }

    int n,k; cin >> n >> k;
    pair<int,int> a[n];


    f0(i,n) {cin >> a[i].first;a[i].second=i;}
    ll ans=a[0].first;

    // a[i].first la gia tri cua so hang thu i cua bieu thuc

    // a[i].second = i, muc dich de giu lai index ban dau cua so hang

    sort(a,a+n,greater<pair<int,int>>());
    int i = 0;
    while(k--){
        if(a[i].second != 0) ans+=a[i].first; // bo qua phan tu dau tien
        else k++;
        i++;
    }
    while(i < n){
        if(a[i].second != 0) ans -= a[i].first; // bo qua phan tu dau tien
        i++;
    }
    cout << ans;
    return 0;
}

File: THCS/2023_2024/LaoCai_2324/Bai3/bai3.py
inp_file_name = "bai3"

def main():
    if inp_file_name != "":
        s = inp_file_name + ".inp"
        freopen_in = open(s, "r")
        s = inp_file_name + ".out"
        freopen_out = open(s, "w")

    # Đọc dòng đầu tiên chứa n và k
    n, k = map(int, freopen_in.readline().split())
    a = []

    # Đọc tiếp các số hạng từ file, lưu vào danh sách
    numbers = list(map(int, freopen_in.readline().split()))  # Đọc tất cả các số từ dòng này

    for i in range(n):
        a.append((numbers[i], i))  # Lưu giá trị và index vào danh sách

    ans = a[0][0]  # a[0][0] là giá trị của số hạng đầu tiên

    # a[i][1] = i, mục đích để giữ lại index ban đầu của số hạng
    a.sort(reverse=True, key=lambda x: x[0])  # Sắp xếp theo giá trị giảm dần

    i = 0
    while k > 0:
        if a[i][1] != 0:
            ans += a[i][0]  # Bỏ qua phần tử đầu tiên
        else:
            k += 1
        i += 1
        k -= 1

    while i < n:
        if a[i][1] != 0:
            ans -= a[i][0]  # Bỏ qua phần tử đầu tiên
        i += 1

    print(ans, file=freopen_out)

    # Đóng file sau khi hoàn thành
    freopen_in.close()
    freopen_out.close()

if __name__ == "__main__":
    main()
